### DISCUSSION TITLE
How to Instanciate an object

### DISCUSSION THREAD

**Discussion by RemiDesgrange** [zhukovgreen/pytest-when#32] [Q&A]
Hi, 

I may have missed something, but I can't find a way to override object instantiation ex:

```python
class MyClass:
    def __init__(self, an_arg):
        pass

class MockedClass:
    def __init__(self, an_arg):
         print('modified behavior')

@pytest.fixture
def some_fixture(when):
     when(MyClass).called_with('some_arg').then_return(MockedClass("something"))
```

Thx.

**Comment by zhukovgreen** [comment]
Hi @RemiDesgrange let me check. Otherwise I was not designed for such use case, because you need to keep the MockedClass and MyClass in sync. Generally you may want to be more granular and overwrite just some method

**Reply by RemiDesgrange** [reply]
thx for the quick reply. That's what I thought. I wanted to override an `HttpResponse`from django called in a view. I can maybe override the `__call__` of HttpResponse
